### PR TITLE
keepassxc: add 'wrapGAppsHook' to fix FileChooser error (see also: https://github.com/NixOS/nixpkgs/issues/111368)

### PR DESCRIPTION
**Pull request by balazs-lengyel**
###### Motivation for this change
KeePassXC failed creating a DB with `GLib-GIO-ERROR **: 11:16:48.748: Settings schema 'org.gtk.Settings.FileChooser' is not installed`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xfix** [comment]
Huh, that's interesting, KeepassXC is not a GTK application, however apparently it's possible to configure Qt applications to use GTK with `QT_QPA_PLATFORMTHEME=gtk3`. That has interesting implications for all Qt applications in Nixpkgs.

**Comment by xfix** [comment]
Seems like https://github.com/NixOS/nixpkgs/pull/136071 could be related, cc @jtojnar.

File: pkgs/applications/misc/keepassx/community.nix [hunk 1]
```diff
@@ -90,7 +91,7 @@ stdenv.mkDerivation rec {
     runHook postCheck
   '';
 
-  nativeBuildInputs = [ asciidoctor cmake wrapQtAppsHook qttools pkg-config ];
+  nativeBuildInputs = [ asciidoctor cmake wrapGAppsHook wrapQtAppsHook qttools pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/keepassx/community.nix:94]: This double wraps the binary if I am not mistaken. Please add the wrapper arguments to wrapperArgs.

**Comment by turion** [comment]
What issue did this fix? I've never heard of this problem.

**Comment by balazs-lengyel** [comment]
@turion keepassxc crashes on selecting the location for the new DB file under a fresh nixos+gnome install.

**Comment by turion** [comment]
That explains why it doesn't have any problems in KDE.

**Comment by turion** [comment]
Why is double wrapping a problem? To me it looks like a more maintainable and clear solution.

**Comment by turion** [comment]
> Double wrapping is a problem because Linux does not support passing setting `argv[0]` when `exec`ing a script. That means `argv[0]` will contain `.program-wrapped`, which will likely get used in `--help` at best, and prevent the program from finding its resources at worst.

Ah, ok. Quite a few (~ 10) programs in nixpkgs do this, though. Since this seems like a popular thing to do, and they all do it wrong, allegedly, maybe there should be a standard "double wrapper" `wrapQtAndGAppsHook` containing both wrapping args.

**Comment by Artturin** [comment]
can't repro the error on master `XDG_DATA_DIRS= QT_QPA_PLATFORMTHEME=gtk3 ./result/bin/keepassx`

**Comment by turion** [comment]
This is my fault. I added double wrapping in https://github.com/NixOS/nixpkgs/pull/163608/, not realizing that there are downsides.

For `keepassxc` at least, everything works sort of as expected:

```
$ keepassxc --help
Usage: /nix/store/bsn1wir6x2jiin6jdp3jzl4ja98b2yxn-keepassxc-2.6.6/bin/.keepassxc-wrapped [options] [filename(s)]
KeePassXC - cross-platform password manager

Options:
  -h, --help                   Displays help on commandline options.
  --help-all                   Displays help including Qt specific options.
  -v, --version                Displays version information.
  --config <config>            path to a custom config file
  --localconfig <localconfig>  path to a custom local config file
  --lock                       lock all open databases
  --keyfile <keyfile>          key file of the database
  --pw-stdin                   read password of the database from stdin
  --debug-info                 Displays debugging information.

Arguments:
  filename(s)                  filenames of the password databases to open
                               (*.kdbx)
```

**Pull request closed without merging**